src/library/export/dlgtrackexport.h

```diff
@@ -0,0 +1,62 @@
+#ifndef DLGTRACKEXPORT_H
+#define DLGTRACKEXPORT_H
+
+#include <future>
+
+#include <QDialog>
+#include <QString>
+#include <QScopedPointer>
+#include <QTime>
+
+#include "configobject.h"
+#include "library/export/trackexport.h"
+#include "library/export/ui_dlgtrackexport.h"
+
+// A dialog for interacting with the track exporter in an interactive manner.
+// Handles errors and user interactions.
+class DlgTrackExport : public QDialog,
```

Comment:
> I want to make trackexport not depend on a GUI to work

I would say this is a requirement for new code. We have too much logic built into our dialogs and other GUI code. I'm in the middle of disentangling this in other parts of Mixxx and already went through the giant mess of doing this with e.g. AutoDJ.